daemon/archive.go

```diff
@@ -236,7 +236,13 @@ func (daemon *Daemon) containerArchivePath(container *container.Container, path
 	if driver.Base(resolvedPath) == "." {
 		resolvedPath += string(driver.Separator()) + "."
 	}
-	sourceDir, sourceBase := driver.Dir(resolvedPath), driver.Base(resolvedPath)
+
+	sourceDir := resolvedPath
+	sourceBase := "."
+
+	if stat.Mode&os.ModeDir == 0 { // not dir
```

Comment:
I think this should be

```
if symlink {
opts = rebase
}
```

and always `archivePath(resolvedPath)`.

Would be more understandable what is actually happening and wouldn't need the hack for removing the slash in the walker. But functionally they seem equivalent.

Reply:
Agreed. Can be done in a follow up.